src/hmem_ze.c

```diff
@@ -37,7 +37,7 @@
 #include "ofi_hmem.h"
 #include "ofi.h"
 
-#if HAVE_LIBZE
+#if HAVE_LIBZE || defined(ENABLE_ZE_DLOPEN)
```

Comment:
This is not aligned with cuda or rocr support.  So, are those broken as well, or is there an issue with ze's configure checks?

Reply:
I think the original code is correct. The issue is with configuring with '--enable-ze-dlopen' on system w/o having ze library on the system.

Reply:
As far as I can tell, all of them should have similar issues though I can't verify it. @wzamazon could you chime in here?
I think the dlopen/with-<dev> configure code for all interfaces could be better written.

Reply:
It's an user error rather than code error.

Reply:
So you do need the library for the dlopen? I was thinking it was just the headers

Reply:
That's true for all of cuda/rocr/ze. The requirement may be more than what is needed, but that's how the configure scripts set the HAVE_xxx macro.

Reply:
Are you suggesting keeping the higher requirement of needing libraries+headers or rewriting the configure scripts to define the HAVE_xxx macro to be have_headers + (have_libs || enable_dl) ?

Reply:
> It's a user error rather than code error.

@aingerson To clarify, this comment was replying to your comment above it. I wrote it before your post appeared :)

Reply:
@aingerson I have no preference with one way over another.